### DISCUSSION TITLE
Camera stream from external video door unit?

### DISCUSSION THREAD

**Discussion by flimofly** [joleys/niko-home-control-II#118] [Q&A]
Has anyone been able to display a video stream from the external video unit ([550-22012](https://www.niko.eu/en/products/niko-home-control/ip-products/external-video-unit-productmodel-niko-9d1cdb0b-fada-4f48-ac40-1d796d583578/550-22011)) in a Home Assistant dashboard?

The stream from entity `camera.external_video_unit_with_1_illuminated_touch_button_inkom` keeps loading but does not display anything.

**Comment by tijsverkoyen** [comment]
Is your camera connected directly to the controller?

> Remark: Note that this will only work if your camera is **not** directly connected to the controller. The controller uses its own DHCP server and there is nothing in place to connect to the streams. In older versions of the controller there was port forwarding, but this is disabled (confirmed by Niko).

See https://github.com/joleys/niko-home-control-II#robinsip-videodoorstation

**Comment by flimofly** [comment]
Thanks it is indeed connected to the POE port on the controller. Sorry that I missed this.

I assume it should be connected to a POE switch. But can the NHC app still see it then?

**Comment by tijsverkoyen** [comment]
I don't own a camera. So I can't test. But I think @joleys has it working?

**Comment by larserhardsen** [comment]
I have the Video door station connected to my HA instance.
I have created a separate VLAN on my switch and wired it to one of the extra ethernet ports on the controller. 
In HA I have added an extra interface with the VLAN attached and in that way the HA instance gets an additional IP via DHCP from the controller.
I liked the port forwarding approach better

**Comment by flimofly** [comment]
Smart. What do you mean by an extra interface in HA? Is this in the NHC2 integration? 

With this setup you can still use the NHC app to see who is at the door and open it?

**Comment by larserhardsen** [comment]
> Smart. What do you mean by an extra interface in HA? Is this in the NHC2 integration?

I run HAOS as a VM and have added an extra network interface to it, which I have attached to a VLAN.
How are you running HA?

> With this setup you can still use the NHC app to see who is at the door and open it?

Yes!

**Comment by flimofly** [comment]
I am running HA in a docker container (Truenas app) so not sure this is possible.

**Comment by Roeli1996** [comment]
> Thanks it is indeed connected to the POE port on the controller. Sorry that I missed this.
> 
> I assume it should be connected to a POE switch. But can the NHC app still see it then?

The controller can still see the camera if it is connected to a POE switch.

I have the following setup:
Everything linked via Ethernet over two Unifi POE switches and then the connection between a switch and the controller.
I'm using an image entity, on live and as an entity it's just the name of my video outpost.

For your information, apart from Home Assistant: you can always consult your outpost via the IP address (local). Then log in with 'user' and a blank password.

**Comment by joleys** [comment]
I moved the door station network connection from the internal NHC Controller network to my general home network.  This way both NHC and HA can access the camera feed.
Note: the touchscreen often fails to display the video feed when someone rings the bell.  This could be something on the NHC side only, not related to the network setup.

**Comment by Roeli1996** [comment]
I have the same setup, but I don't have that problem on the touchscreen.  Or at least not that I've noticed it yet.  What I have noticed is that the video stream in the Niko app is sometimes "black".  After closing and reopening, the video returns.




Arne
